src/routes.rs

```diff
@@ -141,6 +158,13 @@ pub async fn dispense_tokens(
         });
     }?;
 
+    if has_reached_dispense_limit(&dispense_tracker, address, config.dispense_limit_interval).await
+    {
+        return Err(error(
```

Comment:
Continued from my comment below: It looks like the `error` function is what is responsible for returning the 500 here - maybe we can parameterize this function to accept a status code, so that we can return a 429 in this case, and continue to return 500 in areas where it is appropriate.